File: sdkwork-ai-sdk-swift/Sources/API/ImagesApi.swift
import Foundation

public class ImagesApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create image variation
    public func createImageVariation(body: CreateImageVariationRequest? = nil, params: [String: Any]? = nil) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/images/variations"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? ImageGenerationResponse
    }

    /// Create image
    public func createImage(body: ImageGenerationRequest) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/images/generations"), body: body)
        return response as? ImageGenerationResponse
    }

    /// Create image edit
    public func createImageEdit(body: CreateImageEditPostRequest? = nil, params: [String: Any]? = nil) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/images/edits"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? ImageGenerationResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/ImagesApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ImagesApi
    {
        private readonly HttpClient _client;

        public ImagesApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Create image variation
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageVariationAsync(ImageGenerationRequest body)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/images/variations"), body);
        }

        /// <summary>
        /// Create image
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageAsync(ImageGenerationRequest body)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/images/generations"), body);
        }

        /// <summary>
        /// Create image edit
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageEditAsync(ImageGenerationRequest body)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/images/edits"), body);
        }
    }
}
